File: BroadcastReceiver/app/src/main/java/edu/upbc/broadcastreceiver/MainActivity.java
package edu.upbc.broadcastreceiver;

import androidx.appcompat.app.AppCompatActivity;

import android.content.IntentFilter;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        IntentFilter intentFilter = new IntentFilter("edu.upbc.DATA_TRANSFER");
        MyReceiver objReceiver = new MyReceiver();
        registerReceiver(objReceiver, intentFilter);
    }
}

File: BroadcastReceivers/app/src/main/java/org/bedu/broadcastreceivers/MainActivity.kt
package org.bedu.broadcastreceivers


import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import org.bedu.broadcastreceivers.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val bundle = Bundle().apply {
            putString("KEY_NAME", "Martin")
            putString("KEY_EMAIL", "gmlimon@gmail.com")
        }
        binding.button.setOnClickListener {
            CoroutineScope(Dispatchers.IO).launch {
                delay(1000)
                Intent("org.bedu.receiver.DATA_TRANSFER").apply {
                    putExtras(bundle)
                }.let(::sendBroadcast)
            }
            }

        }
}